build: Rebuild for Qt6

effigies opened the pull request:
This PR is intended to allow `quazip` to be used in qt5 or qt6 environments, which will in turn allow downstream tools to provide qt5/qt6 variants.

The motivation here is that I have an environment with a CLI that is built against qt5 and the latest matplotlib is built against qt6, so I can't upgrade. I'm trying to upgrade the CLI feedstock in a similar fashion, and this PR will help there.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

effigies committed:
feat: Add build variants for qt5 and qt6

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ This recipe is using a compiler, which now requires adding a build dependence on `{{ stdlib("c") }}` as well. Note that this rule applies to each output of the recipe using a compiler. For further details, please see https://github.com/conda-forge/conda-forge.github.io/issues/2102.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13706365411](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13706365411). Examine the logs at this URL for more detail.</sub>

effigies committed:
chore(build): Add stdlib

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

effigies committed:
chore(osx): Add OSX build requirements

effigies committed:
build: Disable availability checks for libcpp

effigies committed:
MNT: Re-rendered with conda-build 25.4.2, conda-smithy 3.48.1, and conda-forge-pinning 2025.05.15.14.51.11

effigies commented:
@izahn This is ready to go, if you wouldn't mind taking a look!

effigies commented:
@conda-forge-admin, please rerender

effigies commented:
@izahn I wonder if you have time to look at this?

effigies committed:
Add self as maintainer

effigies commented:
@conda-forge-admin, please rerender

effigies commented:
@conda-forge/core I'm willing to maintain this recipe. I've added myself to the maintainers' list. I am on a number of other recipes: https://github.com/search?q=org%3Aconda-forge%20%2F-%20effigies%2F&type=code

Commit:
MNT: Re-rendered with conda-build 25.4.2, conda-smithy 3.48.1, and conda-forge-pinning 2025.05.22.20.57.09

h-vetinari commented on recipe/conda_build_config.yaml:
```diff
@@ -0,0 +1,11 @@
+MACOSX_SDK_VERSION:        # [osx]
+  - '11.0'                 # [osx]
+
+c_stdlib_version:          # [linux]
+  - 2.17                   # [linux]
```
This is not necessary anymore, as it's not the default.

effigies replied:
I thought if we didn't pin 2.17, then newer glibc will be used, and the packages won't work for users stuck on centos7 (most academic HPC users, AFAICT)?

h-vetinari commented on recipe/conda_build_config.yaml:
```diff
@@ -0,0 +1,11 @@
+c_stdlib_version:          # [osx]
+  - '11.0'                 # [osx]
+MACOSX_SDK_VERSION:        # [osx]
+  - '11.0'                 # [osx]
```
If you raise the stdlib version, then the MACOSX_SDK_VERSION does not need to be specified (it always satisfied >=c_stdlib_version).

h-vetinari commented on recipe/meta.yaml:
```diff
@@ -20,16 +24,20 @@ requirements:
-    - zlib
+    - qt-main                        # [build_variant == "qt5"]
+    - qt6-main                       # [build_variant == "qt6"]
+    # Compatible with 1.2 and 1.3
+    - zlib =1.2                      # [build_variant == "qt5"]
+    - zlib =1.3                      # [build_variant == "qt6"]
```
Why are you building the qt5 variant against zlib 1.2? Everything built against 1.2 is compatible with 1.3 (and we've patched this retroactively), so I don't see why this is necessary. It should stay
```suggestion
    - zlib
```

effigies replied:
I believe this helped with dependency resolution for things built with older versions of zlib 1.2 that had pins of `<1.3`. I will build locally and see if I can install packages with just zlib.

h-vetinari reviewed (commented):
Thanks for stepping up! Two nits (not important), but one question about what you're doing with zlib.

hmaarrfk commented on recipe/meta.yaml:
```diff
@@ -20,16 +24,20 @@ requirements:
     - {{ compiler('cxx') }}
+    - {{ stdlib('c') }}
+    - noqt6                          # [build_variant == "qt5"]
+    - noqt5                          # [build_variant == "qt6"]
```
These noqt stuff are not necessary. Your build dependencies are very simple.

effigies committed:
Simplify build configuration

effigies committed:
MNT: Re-rendered with conda-build 25.1.2, conda-smithy 3.48.1, and conda-forge-pinning 2025.05.23.05.43.47

MNT: Re-rendered with conda-build 25.1.2, conda-smithy 3.48.1, and conda-forge-pinning 2025.05.23.05.43.47

effigies commented:
@h-vetinari @hmaarrfk Thanks for the reviews. I simplified the dependencies and things seem to be working. I will ping you if it seems like I need to complicate them again, if that's alright.

hmaarrfk commented:
one question i have for you is:

do you need both qt5 and qt6 support?

The increase in build matrix and complexity is rather cumbersome to continue.

Why not just move to qt6?

effigies commented:
Unless things have changed, I couldn't install both qt5 and qt6 tools into the same conda environment. Thus, if my dependency tree contained a tool built with qt5 and a tool built with qt6, I couldn't resolve.

So the goal here is to allow tools that depend on quazip to also have qt5 and qt6 builds. I suppose I could move to qt6, and the downstream build would use the older version for qt5 and the newer versions for qt6.

hmaarrfk commented:
> I suppose I could move to qt6, and the downstream build would use the older version for qt5 and the newer versions for qt6.

This is MUCH MUCH simpler to support and I would strongly encourage this.

I moved opencv to drop all qt5 builds.

> Unless things have changed, I couldn't install both qt5 and qt6 tools

We have worked hard to enable this possibility, so i'm surprised that this is what you experienced. However, it is almost not worth troubleshooting IMO.

effigies committed:
chore: Drop qt5 variant

effigies committed:
MNT: Re-rendered with conda-build 25.1.2, conda-smithy 3.48.1, and conda-forge-pinning 2025.05.23.05.43.47

effigies commented:
@hmaarrfk Done.
